In [44]:
from autoscraper import AutoScraper

url = 'https://cancerstatisticscenter.cancer.org/#!/cancer-site/Non-Hodgkin%20lymphoma'
url = 'https://cancerstatisticscenter.cancer.org/module/c23c2QoY'
url = 'https://acsjournals.onlinelibrary.wiley.com/doi/full/10.3322/caac.21590'
wanted_dict = {'State': ["Alaska"], 'Estimated new cases, 2020': ["120"]} # this pull a large text
wanted_dict = {'STATE': ["Alabama"], 'NON‐HODGKIN LYMPHOMA': ['290']} # this pull a large text

#wanted_list = ["NON‐HODGKIN LYMPHOMA"]
wanted_list = ['STATE']
scraper = AutoScraper()
scraper.build(url, wanted_dict=wanted_dict)
#scraper.build(url, wanted_list)

#result = scraper.get_result_similar(url, group_by_alias=True)
result = scraper.get_result_similar(url, group_by_alias=False)

print(result)

['All sites', 'Oral cavity & pharynx', 'Digestive system', 'Respiratory system', 'Bones & joints', 'Soft tissue (including heart)', 'Skin (excluding basal & squamous)', 'Breast', 'Genital system', 'Urinary system', 'Eye & orbit', 'Brain & other nervous system', 'Endocrine system', 'Lymphoma', 'Myeloma', 'Leukemia', 'Other & unspecified primary sites\ncc \n                                    More deaths than cases may reflect a lack of specificity in recording the underlying cause of death on death certificates and/or an undercount in the case estimate.', 'Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California', 'Colorado', 'Connecticut', 'Delaware', 'Dist. of Columbia', 'Florida', 'Georgia', 'Hawaii', 'Idaho', 'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico', 'New York', 'North Carolina', 'North Dakota', 

In [38]:
type(result)

list

In [15]:
result2 = result
print(result2)

{'State': ['All sites', 'All sites', 'All sites', 'All sites', 'Oral cavity & pharynx', 'Oral cavity & pharynx', 'Oral cavity & pharynx', 'Oral cavity & pharynx', 'Digestive system', 'Digestive system', 'Digestive system', 'Digestive system', 'Respiratory system', 'Respiratory system', 'Respiratory system', 'Respiratory system', 'Bones & joints', 'Bones & joints', 'Bones & joints', 'Bones & joints', 'Soft tissue (including heart)', 'Soft tissue (including heart)', 'Soft tissue (including heart)', 'Soft tissue (including heart)', 'Skin (excluding basal & squamous)', 'Skin (excluding basal & squamous)', 'Skin (excluding basal & squamous)', 'Skin (excluding basal & squamous)', 'Breast', 'Breast', 'Breast', 'Breast', 'Genital system', 'Genital system', 'Genital system', 'Genital system', 'Urinary system', 'Urinary system', 'Urinary system', 'Urinary system', 'Eye & orbit', 'Eye & orbit', 'Eye & orbit', 'Eye & orbit', 'Brain & other nervous system', 'Brain & other nervous system', 'Brain & 

In [3]:
from autoscraper import AutoScraper

#url = 'https://cancerstatisticscenter.cancer.org/#!/data-analysis/NewCaseEstimates'
url = 'https://cancerstatisticscenter.cancer.org/module/c23c2QoY'
#wanted_list_state = ["California"]
scraper = AutoScraper()
result_statename = scraper.build(url, wanted_list_state)
print(result_statename)

None


In [18]:
import bs4 as bs
import urllib.request

#sauce = urllib.request.urlopen('https://cancerstatisticscenter.cancer.org/cancer-site/Non-Hodgkin%20lymphoma/THdtY83c').read()
sauce = urllib.request.urlopen('https://acsjournals.onlinelibrary.wiley.com/doi/full/10.3322/caac.21590').read()
URL = 'https://acsjournals.onlinelibrary.wiley.com/doi/full/10.3322/caac.21590'

page = requests.get(URL)
soup = bs.BeautifulSoup(page.content, 'html.parser');

HTTPError: HTTP Error 403: Forbidden

In [6]:
# find navigation
nav = soup.nav

In [21]:
import requests
page = requests.get('https://acsjournals.onlinelibrary.wiley.com/doi/full/10.3322/caac.21590')
page

<Response [403]>

It is odd while Autoscraper works, requests give 403 (forbidden)